library/Vanilla/Web/JsonView.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * @author Alexandre (DaazKu) Chouinard <alexandre.c@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license https://opensource.org/licenses/GPL-2.0 GPL-2.0
+ */
+
+namespace Vanilla\Web;
+
+use Garden\Web\ViewInterface;
+use Garden\Web\Data;
+
+/**
+ * Class JsonView
+ */
+class JsonView implements ViewInterface {
+
+    /** @var WebLinking */
+    private $webLinking;
+
+    /**
+     * JsonView constructor.
+     *
+     * @param WebLinking $webLinking
+     */
+    public function __construct(WebLinking $webLinking) {
+        $this->webLinking = $webLinking;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function render(Data $data) {
+
+        $paging = $data->getMeta('paging', []);
+
+        foreach(($paging['links'] ?? []) as $relType => $uri) {
+            $this->webLinking->addLink($relType, $uri);
+        }
+
+        $data->setHeader('Link', $this->webLinking->getLinkHeaderValue());
```

Comment:
What if the data has other link headers? Think about how a developer might want to extend this.

Reply:
They just have to call "add" on the webLinking object.

Reply:
We will have to add logics in here to handle that.

Reply:
Okay do it.

Reply:
Done already: https://github.com/vanilla/vanilla/pull/6537/commits/f90b1d31679574039619cc47b61262341b429015